app.ts

```diff
@@ -0,0 +1,238 @@
+
+		//         console.log("Error 404");
+		//     }
+		// };
+		// }
+		//region Data Loading methods
+		async function LoadRecordsData(fromID: number, toID: number): Promise<number[]> {
+			try {
+				const recordsvalue = await getRecordsCall(fromID, toID)
+				let DisplayContent = '';
+				for (const record of recordsvalue) {
+					DisplayContent += `<tr id="table-row-${record[0]}">`;
+					for (const column of record) {
+						DisplayContent += `<td align="center">${column}</td>`;
+					}
+					DisplayContent += '</tr>';
+					$("#wrapper-table-content-body").empty();
+					$("#wrapper-table-content-body").append(DisplayContent);
+				}
+			} catch (error) {
+				console.log("Error <No data>" + error)
+			}
+			return [fromID, toID];
+		}
+		function RecordsFromCursor(cursor: number[]): Promise<number[]> {
+			cursor = cursor.sort((a, b) => { return a - b });
+			return LoadRecordsData(cursor[0], cursor[1]);
+			// throw new Error("Error");
+		}
+		window.onresize = () => {
+			try {
+				const nextToId = calculateToId(previous[0]);
+				clearTimeout(Load);
+				Load = setTimeout(async () => {
+					const recordCount = await getRecordCountCall();
+					if (nextToId >= recordCount - 1) {
+						const fromId = recordCount - 1 - (calculateToId(previous[0]) - previous[0]);
+						const toId = recordCount - 1;
+						previous = await LoadRecordsData(fromId, toId);
+					} else {
+						previous = await LoadRecordsData(previous[0], nextToId)
+					}
+				}, 10);
+			} catch (error) {
+				// throw new Error("Error" + error);
+			}
+		}
+		// Handlers
+		async function LoadPageContent(fromID: number, toID: number): Promise<number[]> {
+
+			try {
+				let DisplayContent = "";
+				const columns = await getColumnNamesCall();
+				for (const column of columns) {
+					DisplayContent += `<th align="center">${column}</th>`;
+					$("#wrapper-table-header-row").empty();
+					$("#wrapper-table-header-row").append(DisplayContent);
+				}
+			} catch (error) {
+				console.log("Error <No Data>" + error);
+			}
+			return await LoadRecordsData(fromID, toID);
+		}
+		function ConvertNumber(input: string | number, parseAsInt: boolean = true): number {
+			switch (typeof input) {
+				case ('string'):
+					if (parseAsInt == true) {
+						return parseInt(input as string);
+					}
+					return parseFloat(input as string);
+				case ("number"):
+					return input as number;
+				default:
+					return 0;
+			}
+			throw new Error("Error");
+		}
+		function calculateToId(fromId: number): number {
+			const possibleRecordsData = Math.floor((window.innerHeight - ($("#form-content").innerHeight() as number)) / 37);
+			const possibleId = fromId + possibleRecordsData;
+
+			let recordDisplayset = 0;
+			switch (recordDisplayset) {
+				case 0:
+					window.innerHeight <= 646;
+					break;
+				case 1:
+					window.innerHeight <= 969;
+					break;
+				case 2:
+					window.innerHeight <= 1938;
+					break;
+				default:
+					recordDisplayset = 15;
+					break;
+			}
```

Comment:
The result of this bit of code will always have recordDisplayset set to 0... im not sure what your trying to achive here. but the hardcoding for handling diffrent window heights is not what you should be doing. Why is `possibleId` on its own not enough?

Reply:
The first time when I test the calculateToId Function its was not handling different window heights. that is why I possibleId to determine the window heights.

Reply:
No, I agree with that, I just don't get why you add the recordDisplayset to the possibleId? And either way, if I removed the lines I commented about then it would have no affect on the result, so I dont know why you have them. You should really remove them.